blocks/library/gallery/index.js

```diff
@@ -61,16 +70,11 @@ registerBlockType( 'core/gallery', {
 			query( 'div.blocks-gallery figure.blocks-gallery-image img', {
 				url: attr( 'src' ),
 				alt: attr( 'alt' ),
-			} ),
+			} ) || [],
```

Comment:
>I still think it would make the most sense the block to be a class with few lifecycle methods.

One of my concerns here is difficulty in extending base classes if a plugin author chooses not to use ES6. This can be seen in React's `createClass` deprecation, which will now require an additional dependency if developers are bound to it for whatever reason. Class inheritance is not entirely unwieldy in ES5, depending if we need extends vs. plain class.

ES5 `class TextBlock`:

```js
function TextBlock() {
	// Constructor
}

TextBlock.prototype.edit = function() {
	// Editor form
};
```

ES5 `class TextBlock extends Block`:

```js
function TextBlock() {
	Block.apply( this, arguments );

	// Constructor
}

// Extends
TextBlock.prototype = Object.create( Block.prototype );
TextBlock.prototype.constructor = TextBlock;

TextBlock.prototype.edit = function() {
	// Editor form
};
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create#Classical_inheritance_with_Object.create()